### PR TITLE
Add tutorial for Resnet cross platform inference

### PR DESCRIPTION
**Pull request by KevinH48264**
Description: Adding a tutorial for ORT Inferencing on CPU, GPU, and OpenVINO

Motivation and Context
It teaches developers how to use ORT for inferencing on CPU, GPU, and OpenVINO

Staging link: https://natke.github.io/onnxruntime/

### CODE REVIEW

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
+---
+nav_exclude: true
```
Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:2]: Why exclude from the nav?

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+2. convert to ONNX, and
+3. deploy to the default CPU, NVIDIA CUDA (GPU), and Intel OpenVINO with ONNX Runtime -- using the same application code to load and execute the inference across hardware platforms.
+
+[ONNX](https://onnx.ai/) was developed as the open-sourced ML model format by Microsoft, Meta, Amazon, and other tech companies to standardize and make it easy to deploy Machine Learning models on various types of hardware.[ONNX Runtime](https://onnxruntime.ai/) was contributed and is maintained by Microsoft to optimize ONNX model performance over frameworks like PyTorch, Tensorflow, and more. The ResNet-50 model is a commonly used model for image classification that is pretrained on ImageNet.
+
+This tutorial demonstrates how to run an ONNX model on CPU, GPU, and OpenVINO hardware with ONNX Runtime, using [Microsoft Azure Machine Learning](https://azure.microsoft.com/en-us/services/machine-learning/).
+
+## Setup
+
+- **OS Prerequisities:** Your environment should have [curl](https://curl.se/download.html) installed. This tutorial uses a Windows OS.
```
Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:23]: I don't think there is anything specific to Windows in this tutorial. If that's the case we can remove the bit about Windows

Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:23]: I suggest removing the bullets - you've already boldened the headings and I think without the bullets, the format/layout would look better

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+- **Device Prerequisites** : The onnxruntime-gpu library needs access to a [NVIDIA CUDA](https://docs.nvidia.com/cuda/cuda-installation-guide-microsoft-windows/index.html#:~:text=You%20can%20verify%20that%20you,that%20GPU%20is%20CUDA%2Dcapable.) accelerator in your device or compute cluster, but running on just CPU works for the CPU and OpenVINO-CPU demos.
+- **Inference Prerequisites:** Ensure that you have an image to inference on. For this tutorial, we have a "cat.jpg" image located in the same directory as the Notebook files.
+- **Environment Prerequisites:** In Azure Notebook Terminal or AnaConda prompt window, run the following commands to create your 3 environments for CPU, GPU, and/or OpenVINO (differences are bolded).
```
Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:26]: Indent these so that it is clear that they are inside the section about environment

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+*CPU*
+```python
```
Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:29]: These should all be console, not python

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+conda install -c anaconda ipykernel
+conda install -c conda-forge ipywidgets
+python -m ipykernel install --user --name=openvino_env_demo
+python -m pip install --upgrade pip
+pip install openvino
+```
+
+- **Library Requirements:** In the first code cell, install the necessary libraries with the following code snippets (differences are bolded).
+
+*CPU + GPU*
```
Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:61]: Indent these subsections

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+plastic bag 0.0056432663
+OpenVINO CPU Inference time = 31.83 ms
+***** Verifying correctness *****
+PyTorch and ONNX Runtime output 0 are close: True
+PyTorch and ONNX Runtime output 1 are close: True
+```
+
+## Conclusion
+
+We've demonstrated that ONNX Runtime is an effective way to run your PyTorch or ONNX model on CPU, NVIDIA CUDA (GPU), and Intel OpenVINO (Mobile). ONNX Runtime enables deployment to more types of hardware that can be found on [Execution Providers](https://onnxruntime.ai/docs/execution-providers/CoreML-ExecutionProvider.html). As you try ONNX Runtime, feel free to leverage the onnxruntime.ai [documentation and tutorials](https://onnxruntime.ai/docs/), and we'd love to hear your feedback by participating in our ONNX Runtime [Github repo](https://github.com/microsoft/onnxruntime).
```
Comment (natke) [inline docs/tutorials/accelerate-pytorch/resnet-inferencing.md:292]: We don't need the backlink to onnxruntime.ai/docs any more

**Review by natke — COMMENTED** [review]
Looks good.

Added a few comments, mostly formatting. For each section make sure the code is indented inside the section. I would also remove the bullets for your section heading, as I think this would make the presentation less cluttered.

File: docs/tutorials/accelerate-pytorch/resnet-inferencing.md (hunk 1 above)
Comment (KevinH48264) [reply docs/tutorials/accelerate-pytorch/resnet-inferencing.md:2]: Not sure, just followed other examples, but I can remove it

**Review by natke — APPROVED** [review]

**Comment by yf711** [comment]
Hi It seems the section **[Comparison with PyTorch](https://onnxruntime.ai/docs/tutorials/accelerate-pytorch/resnet-inferencing.html#comparison-with-pytorch)** has the same code as **[Comparison with OpenVINO](https://onnxruntime.ai/docs/tutorials/accelerate-pytorch/resnet-inferencing.html#comparison-with-openvino)**

**Comment by KevinH48264** [comment]
Hi,

Here is the section for Comparison with PyTorch which should be there
instead.


   1.

   Comparison with PyTorch


# PyTorch Inferencing on ResNet50

import time

latency = []

with torch.no_grad():

    start = time.time()

    pt_output = resnet50(input_batch)

    latency.append(time.time() - start)

# Tensor of shape 1000, with confidence scores over Imagenet's 1000 classes

# The output has unnormalized scores. To get probabilities, you can run a
softmax on it.

probabilities = torch.nn.functional.softmax(pt_output[0], dim=0)

# Show top categories per image

top5_prob, top5_catid = torch.topk(probabilities, 5)

for i in range(top5_prob.size(0)):

    print(categories[top5_catid[i]], top5_prob[i].item())

print("PyTorch CPU/GPU Inference time = {} ms".format(format(sum(latency) *
1000 / len(latency), '.2f')))

print("***** Verifying correctness *****")

for i in range(2):

    print('PyTorch and ONNX Runtime output {} are close:'.format(i),
np.allclose(ort_output, pt_output.cpu(), rtol=1e-05, atol=1e-04))

Sample output:

Egyptian cat 0.7860558032989502

tabby 0.1173100471496582

tiger cat 0.020089421421289444

Siamese cat 0.011728067882359028

plastic bag 0.005217487458139658

PyTorch CPU Inference time = 58.14 ms

***** Verifying correctness *****

PyTorch and ONNX Runtime output 0 are close: True

PyTorch and ONNX Runtime output 1 are close: True


Best,
Kevin
